metricbeat/module/kafka/broker.go

```diff
@@ -174,12 +176,12 @@ func (b *Broker) PartitionOffset(
 	req.AddBlock(topic, partition, time, 1)
 	resp, err := b.broker.GetAvailableOffsets(req)
 	if err != nil {
-		return -1, err
+		return -1, errors.Wrap(err, "get available offsets failed")
 	}
 
 	block := resp.GetBlock(topic, partition)
 	if len(block.Offsets) == 0 {
-		return -1, nil
+		return -1, errors.Wrap(block.Err, "block offsets is empty")
```

Comment:
I am worried this can change some behaviour. But I don't think so, at the end an error was reported by the metricset on this case.